src/main/java/dev/jbang/source/RunContext.java

```diff
@@ -245,6 +247,12 @@ public void setCatalog(File catalogFile) {
 		this.catalogFile = catalogFile;
 	}
 
+	public Properties getContextProperties() {
+		Properties p = new Properties(System.getProperties());
```

Comment:
Is this what you meant to do here? 
I had no idea how `Properties` worked but this code does something very unexpected for somebody (like me) who doesn't.
This actually results in an *empty* Properties object!

I actually find it completely mystifying how this passes the tests. If I run them all there's no error. But when I run the (javaFX) tests one by one, the one using properties fail.

Reply:
Aaaah. That makes sense! Need the detector wired in to the properties rather than system properties. Totally agree. 

And yes it's intended the properties are empty - the system properties just used as fall back.